### ISSUE TITLE
FAILURE: c_macos_dbg_native.bins/dbg/window_overflow_bad_client_test GRPC_POLL_STRATEGY=poll

### ISSUE THREAD

**Issue by apolcyn** [grpc/grpc#12138]
https://grpc-testing.appspot.com/job/gRPC_pull_requests_macos/7320/testReport/junit/(root)/c_macos_dbg_native/bins_dbg_window_overflow_bad_client_test_GRPC_POLL_STRATEGY_poll/

```
.
.
.
.
.
.
.
...............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................'
D0809 14:06:19.368825000 140735171171072 ev_posix.c:121]               Using polling engine: poll
D0809 14:06:19.385875000 140735171171072 dns_resolver.c:316]           Using native dns resolver
E0809 14:06:24.437121000 140735171171072 bad_client.c:161]             assertion failed: gpr_event_wait(&a.done_write, grpc_timeout_seconds_to_deadline(5))



*******************************
Caught signal SIGABRT
  no backtrace

2017-08-09 14:06:37,633 FAILED: bins/dbg/window_overflow_bad_client_test GRPC_POLL_STRATEGY=poll [ret=-6, pid=7583]
```

**Comment by jtattermusch** [comment]
Also seen on master.
https://grpc-testing.appspot.com/job/gRPC_pull_requests_macos/7814/console

**Comment by jtattermusch** [comment]
Seen on master again:
https://sponge.corp.google.com/target?id=67df7321-762c-4936-bba0-ab253c80689f&target=github/grpc/c_macos_dbg_native&searchFor=&show=FAILED&sortBy=STATUS

**Comment by apolcyn** [comment]
seen again on master in opt config:
```
E0912 15:06:20.122867000 140735153025792 bad_client.c:139]             assertion failed: gpr_event_wait(&a.done_write, grpc_timeout_seconds_to_deadline(5))
```

https://grpc-testing.appspot.com/job/gRPC_pull_requests_macos/7869/testReport/junit/(root)/c_macos_opt_native/bins_opt_window_overflow_bad_client_test_GRPC_POLL_STRATEGY_poll/

**Comment by jtattermusch** [comment]
Looks like this is failing pretty consistently, lifting to P1.

**Comment by dgquintas** [comment]
This seems to fail 100% of the time now. Example failure log: https://sponge.corp.google.com/target?id=8337b04e-fe03-4e2c-9666-c7cf074b6ccf&target=grpc/macos/master/grpc_basictests_opt&searchFor=&show=FAILED&sortBy=STATUS

**Comment by yang-g** [comment]
Should be fixed by #12729.
